packages/tokens/src/color/color.types.ts

```diff
@@ -0,0 +1,37 @@
+const State = {
+  Default: 'default',
+  Hover: 'hover',
+  Focus: 'focus',
+} as const;
+
+type State = (typeof State)[keyof typeof State];
+
+const Type = {
+  Background: 'background',
+  Border: 'border',
+  Icon: 'icon',
+  Text: 'text',
+} as const;
+
+type Type = (typeof Type)[keyof typeof Type];
+
+const Variant = {
+  Primary: 'primary',
+  Secondary: 'secondary',
+  Inverse: 'inverse',
+  InversePrimary: 'inversePrimary',
+  InverseSecondary: 'inverseSecondary',
+  Info: 'info',
+  Warning: 'warning',
+  Error: 'error',
+  Success: 'success',
+  Disabled: 'disabled',
+} as const;
+
+type Variant = (typeof Variant)[keyof typeof Variant];
+
+export { State, Type, Variant };
+
+export type MapVariantsToStates = {
+  [k in Variant]?: { [k in State]: string };
+};
```

Comment:
I think this approach is fine 👍 Not sure we need to be super explicit here with which variants have what states

Reply:
Technically could be `Record<Variant, Record<State, string>>`

Reply:
Kk - are you aligned to merge?

Reply:
It can't be that because its a patial of Variant, and Record<Variant was complaining about missing keys

Reply:
and `Partial<Variant` didn't work either